File: py/1422.py
class Solution:
    def maxScore(self, s: str) -> int:
        onesToRight = s.count('1')
        zerosToLeft = 0
        maxScore = 0
        for i in range(len(s) - 1):
            x = s[i]
            if x == '0':
                zerosToLeft += 1
            else:
                onesToRight -= 1
            maxScore = max(maxScore, zerosToLeft + onesToRight)
        return maxScore

File: go/1422.go
func maxScore(s string) int {
    onesToRight := 0
    for _,c := range s {
      if c == '1' {
        onesToRight++
      }
    }
    zerosToLeft, maxScore := 0,0

    for i:=0; i < len(s)-1; i++ {
        if s[i] == '0' {
          zerosToLeft++
        } else {
          onesToRight--
        }
        if maxScore < (onesToRight + zerosToLeft) {
          maxScore = onesToRight + zerosToLeft
        }
    }
    return maxScore
}
